src/api/app/helpers/webui/package_helper.rb

```diff
@@ -103,9 +103,15 @@ def expand_diff?(filename, state)
       (filename == '_patchinfo' || filename.ends_with?('.spec') || filename.ends_with?('.changes'))
   end
 
-  def calculate_revision_on_state(revision, state)
-    result = revision.to_i
-    result -= 1 if state == 'deleted'
-    [result, 0].max
+  def viewable_file?(filename)
+    !Package.is_binary_file?(filename) && !filename.include?('/')
+  end
+
+  def previous_revision(revision)
+    # rubocop:disable Style/RescueModifier
+    is_integer = (Integer(revision) rescue false)
+    # rubocop:enable Style/RescueModifier
+
+    revision.to_i - 1 if is_integer
```

Comment:
what about writing this as `revision.to_i - 1 if revision.present?`

Reply:
I re-arranged the code and dropped this method again. We could only run into this via request diffs. And this is taken care of differently now.

Reply:
but if it is present is always an integer... if not you are doing something wrong somewhere else....

Reply:
I rewrote the code around this (rendering the partials from the package diff and request diff controller). So this code is gone.

Though your statement is wrong "it" is not always an integer. Revisions can be hashes. That's why there is the `@linkinfo` variable.